### PR TITLE
Updated to JSON data

### PR DESCRIPTION
**Pull request by shereenElSayed**
## Description
There are 2 changes: 
### Changing astronotes from csv to json
Changing astronotes extraction from TNS method from csv to json. The JSON format retrieved from TNS contains all the data required.
We didn't use the JSON before bec. it had a bug of empty values for multiple fields. 

### Adding TNS object ID 
The database team required to have the TNS object ID in the JSON data. Currently, the TNS API does not contain this ID, but if you retrieved the data as a CSV file, you can get this ID. So for now, as a workaround until they add this ID to the API, we get the CSV for the specified object and get its ID. (this fix will probably be changed when they make the API return the ID as well) 

Since this is a prototype, I updated the script run every day for the objects and Astronotes. (we will need to make a stream but they don't have this for now)

## Checklist

* [x] All new functions and classes are documented and adhere to Google doc style (3.8.3-3.8.6 of [this](http://google.github.io/styleguide/pyguide.html#383-functions-and-methods) document)
* [ ] Add/update sphinx documentation with any relevant changes.
* [x] Add/update pytest-style tests in `/tests`, ensuring sufficient code coverage.
* [x] `make test` runs without errors.
* [x] `make lint` doesn't give any warnings.
* [x] `make format` doesn't give any code formatting suggestions.
* [ ] `make doc` runs without errors and generated docs render correctly.
* [x] Check that CI pipeline run on this PR passes all stages.
* [ ] Review signoff by at least one developer.

NOTE: If this PR relates to a release, open and reference an issue with the Release checklist template.

### CODE REVIEW

File: stream2hop/tns2hop.py [hunk 1]
```diff
@@ -133,7 +134,22 @@ def fix_spectra(object, parameters_list):
...
+    params = {"name": object_name, "format": "csv"}
+    response = requests.get(url, params=params, stream=True)
+    csv_reader = csv.reader(StringIO(response.content.decode("utf-8")), delimiter=",")
+    i = 0
+    for row in csv_reader:
+        if i == 1:
+            return row[0]
+        i += 1
+
+    return 0
```
Comment (myNameIsPatrick) [inline stream2hop/tns2hop.py]: Is this checking if there's at least 1 row populated in the csv-formatted input. I assume you're skipping over the header here.

What about:
```
next(csv_reader) # remove header
if csv_reader:
    return next(csv_reader)[0]
else:
    return 0
```

By the way, is it always guaranteed to give only zero or one rows?

File: stream2hop/tns2hop.py [hunk 2]
```diff
@@ -205,29 +221,24 @@ def get_astronotes(hop_url, config, api_key):
...
-        if line_count == 0:
-            for headline in row:
-                headlines.append(headline)
-            line_count += 1
-        else:
-            content = {headline: col for headline, col in zip(headlines, row)}
-            json_object = {"format": "BLOB", "content": content}
+    for astronote in astronotes_list.items():
+        if astronote[1]:
+            json_object = {"format": "BLOB", "content": astronote[1]}
```
Comment (myNameIsPatrick) [inline stream2hop/tns2hop.py:228]: This will likely be simpler once the hop-client update is merged in, but we can leave that alone for now.

**Review by myNameIsPatrick — CHANGES_REQUESTED** [review]
Looks good, I just had one minor suggestion but otherwise should be good to go.

File: stream2hop/tns2hop.py (hunk 1 above)
Comment (shereenElSayed) [reply stream2hop/tns2hop.py]: I just wanted to skip the header.
Yes, it's always one 0 or 1 rows, and 99.9% it's 1. The TNS object number I am using in the search is unique.

I will update my code.

File: stream2hop/tns2hop.py (hunk 2 above)
Comment (shereenElSayed) [reply stream2hop/tns2hop.py:228]: yes, I noticed you changed this in the other PR, we will have to update this once the 2 PRs merge.

**Comment by shereenElSayed** [comment]
@myNameIsPatrick Thanks a lot for the review

**Pull request merged**